src/Microplate.php

```diff
@@ -0,0 +1,198 @@
+<?php declare(strict_types=1);
+
+namespace Mll\Microplate;
+
+use Illuminate\Support\Collection;
+use Mll\Microplate\Enums\FlowDirection;
+use Mll\Microplate\Exceptions\MicroplateIsFullException;
+use Mll\Microplate\Exceptions\UnexpectedFlowDirection;
+use Mll\Microplate\Exceptions\WellNotEmptyException;
+
+/**
+ * @template TWell
+ *
+ * @phpstan-type WellsCollection Collection<array<string, TWell>>
+ */
+class Microplate
+{
+    public const EMPTY_WELL = null;
+
+    public CoordinateSystem $coordinateSystem;
+
+    /**
+     * @var WellsCollection
+     */
+    public Collection $wells;
+
+    public function __construct(CoordinateSystem $coordinateSystem)
+    {
+        $this->coordinateSystem = $coordinateSystem;
+
+        $this->clearWells();
+    }
+
+    public function position(Coordinate $coordinate, FlowDirection $direction): int
+    {
+        /** @var int $rowIndex Must be found, since $coordinate->row enforces is valid */
```

Comment:
No longer true, since Coordinate is no longer parameterized by `CoordinateSystem` and thus could have `$row` or `$column` outside of the bounds allowed within this Microplate instance.